packages/flutter/lib/src/material/dropdown.dart

```diff
@@ -612,20 +619,21 @@ class _DropdownRoutePage<T> extends StatelessWidget {
     // and all of the items' intrinsic heights are less than kMinInteractiveDimension.
     // Otherwise the initialScrollOffset is just a rough approximation based on
     // treating the items as if their heights were all equal to kMinInteractiveDimension.
-    if (route.scrollController == null) {
-      final _MenuLimits menuLimits = route.getMenuLimits(buttonRect, constraints.maxHeight, selectedIndex);
-      route.scrollController = ScrollController(initialScrollOffset: menuLimits.scrollOffset);
+    if (widget.route.scrollController == null) {
+      final _MenuLimits menuLimits = widget.route.getMenuLimits(widget.buttonRect, widget.constraints.maxHeight, widget.selectedIndex);
+      widget.route.scrollController = ScrollController(initialScrollOffset: menuLimits.scrollOffset);
```

Comment:
Is the scroll controller necessary to be owned by the route? It seems it can be owned by the `_DropdownRoutePage` directly and pass down to `_DropdownMenu` and `_DropdownMenuItemButton`.

Reply:
There are two options:
1. Create, use and dispose own scroll controller, without using widget.route.scrollController
2. If widget.route.scrollController is null, create, use and dispose own scroll controller. May be we need getter _effectiveScrollController for this.

Which option do you suggest?